cloudflare-analytics.sh

```diff
@@ -36,46 +36,58 @@ cloudflareapikey="${CLOUDFLARE_API_TOKEN:="dummy_api_key"}"
 cloudflarezoneid="${CLOUDFLARE_ZONE_ID:="dummy_zone_id"}"
 cloudflarezone="${CLOUDFLARE_ZONE_NAME:="dummy_zone_name"}"
 
+TMP_FILE_LASTRUN_TIME="${cloudflarezone}_lastrun.tmp"
+TMP_FILE_POSTFIX="${cloudflarezone}_data.$$.tmp"
+
+CF_GQL_RESULTS_LIMIT=10000
+CF_GQL_SINCE_MINS="-419"
+CF_GQL_MIN_PERIOD=60 # Grab additional overlapping data to cover lag CF data availability and outtages
+
+CF_DEFAULT_SINCE_MIN="-10080"
+CF_DEFAULT_UNTIL_MIN="0"
+
 ######################################
 # Define this to be "echo" in ENV to switch off non-observer operations.
 : "${ECHO:=""}"
 
-function process_results {
+#####################################################
+# Processing Analytics API results.
+function process_requests_api {
     local output="$1"
-    local count=$(echo "$output" | jq --raw-output ".result.timeseries" | jq length)
 
+    local count=$(echo "$output" | jq length)
     for i in $(seq -s ' ' 0 $((count-1)))
     do
-      process_result "$output" "$i"
+      local values=$(echo $output | jq .[$i])
+      process_request_api "$values"
     done
 }
 
-function process_result {
+function process_request_api {
     local output="$1"
-    local index=${2:0}
-    local data_tmp="${cloudflarezone}_data.$$.tmp"
+    local data_tmp="api_$TMP_FILE_POSTFIX"
```

Comment:
Should this be using mktemp? Should end up in the right place.

Reply:
Sure.... done